src/views/SignTransaction.vue

```diff
@@ -29,13 +29,16 @@ export default class SignTransaction extends Vue {
     @State private keyguardResult!: KSignTransactionResult | Error | null;
 
     public async created() {
-        if (this.$route.name === `sign-transaction-success`) return;
+        if (this.keyguardResult instanceof Error) {
+            this.rpcState.reply(ResponseStatus.ERROR, this.keyguardResult);
+        } else if (this.keyguardResult) return; // Keyguard success is handled in SignTransactionSuccess.vue
```

Comment:
The else condition here covers all successful signature cases. If the keyguardResult is successful, we do not want to restart the Keyguard request. The sending of the tx is handled in the `SignTransactionSuccess` component.

Reply:
Another useful comment ;)